torchrl/envs/transforms/transforms.py

```diff
@@ -4400,6 +4401,132 @@ def __repr__(self) -> str:
         )
 
 
+class UnaryTransform(Transform):
+    """Applies a unary operation on the specified inputs.
+
+    Args:
+        in_keys (sequence of NestedKey): the keys of inputs to the unary operation.
+        out_keys (sequence of NestedKey): the keys of the outputs of the unary operation.
+        fn (Callable, optional): the function to use as the unary operation.
+        output_spec (TensorSpec, optional): the spec of the output of the operation.
+    """
```

Comment:
```suggestion
    """Applies a unary operation on the specified inputs.

    Args:
        in_keys (sequence of NestedKey): the keys of inputs to the unary operation.
        out_keys (sequence of NestedKey): the keys of the outputs of the unary operation.
        fn (Callable): the function to use as the unary operation.
        output_spec (TensorSpec): the spec of the output of the operation.
    """
```

The output spec could perhaps be optional, when transforming the spec, we could use `spec.zero()`, apply `_fn` and then recast the output to a spec (see `torchrl.envs.utils.make_composite_from_td`)

Also, it seems we're transforming all specs but I'm not sure why (perhaps bc we capture the keys using in/out-keys)? What is the spec of one in-key doesn't match the result of another? Maybe `out_spec` could be a composite spec in that case.

Reply:
I like the idea of using `make_composite_from_td`. Not having to specify the output spec would make it easier to use. However,  `make_composite_from_td` assumes all values are unbounded, and maybe the user would want it to be bounded. So I guess we probably would still need to allow the user to specify the output spec if they want?

> it seems we're transforming all specs but I'm not sure why

I figured it would be ideal to allow each of the `in_keys` to be in any of the output specs, rather than just observations. I also figured that it could be annoying if the user had to specify each of the reward/obs/done specs as three separate arguments. So I made UnaryTransform detect whether a particular `in_key` is part of the observation, reward, or done spec. Does that make sense?

> What is the spec of one in-key doesn't match the result of another? Maybe out_spec could be a composite spec in that case.

Good point. I didn't consider that, but we should support that case. Yes, we could allow `output_spec` to be composite and require that it has an entry for each of the `in_keys` in that case. I'll make that change

Reply:
Cool! I didn't really get why we can't automatically build the spec though, having access to the spec from the env should be enough to create some fake data and extrapolate a spec from it no? (At least in the case of Hash transform)

Reply:
I'm on board with creating the spec automatically, for both Hash and UnaryTransform. I was thinking there might be a reason why the user would want to specify it manually, but I don't have any concrete use case, so I suppose it's probably best to remove the `output_spec` argument for now and only add it back in later if we find a need for it